changelog.d/20230109_172604_sirosen_consolidate_endpointish_setattr.md

```diff
@@ -0,0 +1,14 @@
+### Other
+
+* Improve the uniformity of endpoint and collection option parsing.
+** The `--sharing-restrict-paths` option to `globus collection update` now
+   checks for invalid types (non-dict, non-null data)
+** `globus endpoint update` can now null values which previously could not be
+   unset by passing `""` for `--contact-email`, `--contact-info`,
```

Comment:
TODO: rewrite

`globus endpoint update` will now treat the empty string as `null` for the following options: ...